install/kubernetes/helm/istio/charts/mixer/templates/config.yaml

```diff
@@ -216,28 +218,77 @@ spec:
     responseCode: response.code | 0
     responseSize: response.size | 0
     requestSize: request.size | 0
+    requestId: request.headers["x-request-id"] | ""
+    clientTraceId: request.headers["x-client-trace-id"] | ""
     latency: response.duration | "0ms"
     connectionMtls: connection.mtls | false
     userAgent: request.useragent | ""
     responseTimestamp: response.time
-    receivedBytes: request.total_size | connection.received.bytes | 0
-    sentBytes: response.total_size | connection.sent.bytes | 0
+    receivedBytes: request.total_size | 0
+    sentBytes: response.total_size | 0
     referer: request.referer | ""
+    xForwardedFor: request.headers["x-forwarded-for"] | "0.0.0.0"
+  monitored_resource_type: '"global"'
+---
+apiVersion: "config.istio.io/v1alpha2"
+kind: logentry
+metadata:
+  name: tcpaccesslog
+  namespace: {{ .Release.Namespace }}
+spec:
+  severity: '"Info"'
+  timestamp: context.time | timestamp("2017-01-01T00:00:00Z")
+  variables:
+    connectionEvent: connection.event | ""
+    sourceIp: source.ip | ip("0.0.0.0")
+    sourceApp: source.labels["app"] | ""
+    sourcePrincipal: source.principal | ""
+    sourceName: source.name | ""
+    sourceWorkload: source.workload.name | ""
+    sourceNamespace: source.namespace | ""
+    sourceOwner: source.owner | ""
+    destinationApp: destination.labels["app"] | ""
+    destinationIp: destination.ip | ip("0.0.0.0")
+    destinationServiceHost: destination.service.host | ""
+    destinationWorkload: destination.workload.name | ""
+    destinationName: destination.name | ""
+    destinationNamespace: destination.namespace | ""
+    destinationOwner: destination.owner | ""
+    destinationPrincipal: destination.principal | ""
+    protocol: context.protocol | "tcp"
+    connctionDuration: connection.duration | "0ms"
+    connectionMtls: connection.mtls | false
+    receivedBytes: connection.received.bytes | 0
+    sentBytes: connection.sent.bytes | 0
+    totalReceivedBytes: connection.received.bytes_total | 0
+    totalSentBytes: connection.sent.bytes_total | 0
   monitored_resource_type: '"global"'
 ---
 apiVersion: "config.istio.io/v1alpha2"
 kind: rule
 metadata:
-  name: stdio
+  name: stdiohttp
   namespace: {{ .Release.Namespace }}
 spec:
-  match: "true" # If omitted match is true.
+  match: context.protocol == "http" || context.protocol == "grpc"
   actions:
   - handler: handler.stdio
     instances:
     - accesslog.logentry
 ---
 apiVersion: "config.istio.io/v1alpha2"
+kind: rule
+metadata:
+  name: stdiotcp
+  namespace: {{ .Release.Namespace }}
+spec:
+  match: context.protocol == "tcp"
```

Comment:
Please add a comment like: `TODO: match: context.protocol == "tcp" && connection.event == "open"` or file an issue so that we can track proper usage of the access log.

Reply:
We are tracking connectionEvent as a variable as part of tcpaccesslog.. do you still want different handlers based on connectionEvent?

Reply:
we can add it later. for now, filtering should be sufficient.

Reply:
Added bug https://github.com/istio/istio/issues/6945 for this..